pkg/operator/ceph/cluster/osd/create.go

```diff
@@ -368,7 +368,24 @@ func createDaemonOnPVC(c *Cluster, osd OSDInfo, pvcName string, config *provisio
 	updateConditionFunc(c.clusterInfo.Context, c.context, c.clusterInfo.NamespacedName(), k8sutil.ObservedGenerationNotAvailable, cephv1.ConditionProgressing, v1.ConditionTrue, cephv1.ClusterProgressingReason, message)
 
 	_, err = k8sutil.CreateDeployment(c.clusterInfo.Context, c.context.Clientset, d)
-	return errors.Wrapf(err, "failed to create deployment for OSD %d on PVC %q", osd.ID, pvcName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create deployment for OSD %d on PVC %q", osd.ID, pvcName)
+	}
+
+	if c.spec.Network.MultiClusterService.Enabled {
```

Comment:
yes. This won't be triggered again. Next reconcile checks if we already running osd deployment needs to be updated and that is not a two step process because we use the existing osd to get the exported IP and updated the osd deployment with that IP.  But I'll test it once again to be sure.